tests/lsf/test_lsf_driver.py

```diff
@@ -0,0 +1,140 @@
+import os
+import shutil
+import sys
+from argparse import ArgumentParser
+from textwrap import dedent
+
+import pytest
+
+from ert_shared.cli import ENSEMBLE_EXPERIMENT_MODE
+from ert_shared.cli.main import run_cli
+from ert_shared.main import ert_parser
+
+
+def test_run_mocked_lsf_queue(tmpdir, source_root):
+    shutil.copytree(
+        os.path.join(source_root, "test-data", "local", "poly_example"),
+        os.path.join(str(tmpdir), "poly_example"),
+    )
+
+    shutil.copytree(
+        "mocked_jobs",
+        os.path.join(str(tmpdir), "poly_example"),
+        dirs_exist_ok=True,
+    )
+
+    with tmpdir.as_cwd():
+        with open("poly_example/poly.ert", "w") as fh:
+            config_lines = [
+                "JOBNAME poly_%d\n",
+                "QUEUE_SYSTEM LSF\n",
+                "QUEUE_OPTION LSF MAX_RUNNING 10\n",
+                "QUEUE_OPTION   LSF  BJOBS_CMD  mock_bjobs\n",
+                "QUEUE_OPTION   LSF  BSUB_CMD   mock_bsub\n",
+                "RUNPATH poly_out/real_%d/iter_%d\n",
+                "OBS_CONFIG observations\n",
+                "TIME_MAP time_map\n",
+                "NUM_REALIZATIONS 10\n",
+                "MIN_REALIZATIONS 1\n",
+                "GEN_KW COEFFS coeff.tmpl coeffs.json coeff_priors\n",
+                "GEN_DATA POLY_RES RESULT_FILE:poly_%d.out REPORT_STEPS:0 INPUT_FORMAT:ASCII\n",
+                "INSTALL_JOB poly_eval POLY_EVAL\n",
+                "SIMULATION_JOB poly_eval\n",
+            ]
+
+            fh.writelines(config_lines)
+
+        parser = ArgumentParser(prog="test_main")
+        parsed = ert_parser(
+            parser,
+            [
+                ENSEMBLE_EXPERIMENT_MODE,
+                "poly_example/poly.ert",
+            ],
+        )
+
+        run_cli(parsed)
+
+
+def test_mock_bsub_fail_random(tmpdir, source_root):
+    shutil.copytree(
+        os.path.join(source_root, "test-data", "local", "poly_example"),
+        os.path.join(str(tmpdir), "poly_example"),
+    )
+
+    shutil.copytree(
+        "mocked_jobs",
+        os.path.join(str(tmpdir), "poly_example"),
+        dirs_exist_ok=True,
+    )
+
+    """ 
+    Approx 50% of the submits will fail due to the random generator in the 
+    created mocked bsub script. By using the retry functionality towards 
+    queue-errors in job_queue.cpp we should still manage to finalize all our runs 
+    before exhausting the limits
+    """
+
+    script = dedent(
+        """#!/usr/bin/env python3
```

Comment:
To not having to maintain two submission implementations I suggest we make the fail script only take responsibility for failing and then afterwards calling the `mock_bsub` script above.